pkg/virt-controller/watch/vm/vm_test.go

```diff
@@ -5402,6 +5413,62 @@ var _ = Describe("VirtualMachine", func() {
 					Expect(cond.Status).To(Equal(k8sv1.ConditionTrue))
 					Expect(cond.Message).To(ContainSubstring("invalid volumes to update with migration:"))
 				})
+
+				DescribeTable("should return an error", func(dvExist bool, err error) {
+					testutils.UpdateFakeKubeVirtClusterConfig(kvStore, &v1.KubeVirt{
+						Spec: v1.KubeVirtSpec{
+							Configuration: v1.KubeVirtConfiguration{
+								VMRolloutStrategy: &liveUpdate,
+							},
+						},
+					})
+					virtFakeClient.PrependReactor("patch", "virtualmachineinstance",
+						func(action testing.Action) (handled bool, ret runtime.Object, err error) {
+							fmt.Printf("XXX patch\n")
+							a := action.(testing.PatchActionImpl)
+							patch, err := jsonpatch.DecodePatch(a.Patch)
+							Expect(err).ToNot(HaveOccurred())
+
+							vmi := &v1.VirtualMachineInstance{}
+							obj, err := json.Marshal(vmi)
+							Expect(err).ToNot(HaveOccurred())
+
+							obj, err = patch.Apply(obj)
+							Expect(err).ToNot(HaveOccurred())
+
+							vmi = &v1.VirtualMachineInstance{}
+							Expect(json.Unmarshal(obj, vmi)).To(Succeed())
+							Expect(virtcontroller.NewVirtualMachineInstanceConditionManager().GetCondition(vmi,
```

Comment:
question: cant you call Get of the vmi after the handleVolumeUpdateRequest and the vmi will be recieved with the patch and then check the condition? like you do for the vm:
					Expect(virtcontroller.NewVirtualMachineConditionManager().GetCondition(vm,
						v1.VirtualMachineRestartRequired)).Should(BeNil())

Reply:
No because the vmi is patched and with the condition while the restart require condition is set directly in the vm object (and in the real code updated later).